src/backend/distributed/utils/citus_nodefuncs.c

```diff
@@ -307,3 +329,53 @@ citus_extradata_container(PG_FUNCTION_ARGS)
 
 	PG_RETURN_NULL();
 }
+
+
+#if (PG_VERSION_NUM >= 90600)
+
+/* *INDENT-OFF* */
+#define DEFINE_NODE_METHODS(type) \
+	{ \
+		#type, \
+		sizeof(type), \
+		NULL, \
+		NULL, \
+		Out##type, \
+		Read##type \
+	}
+
+/* *INDENT-ON* */
+const ExtensibleNodeMethods nodeMethods[] =
+{
+	DEFINE_NODE_METHODS(MultiPlan),
+	DEFINE_NODE_METHODS(Job),
+	DEFINE_NODE_METHODS(ShardInterval),
+	DEFINE_NODE_METHODS(MapMergeJob),
+	DEFINE_NODE_METHODS(ShardPlacement),
+	DEFINE_NODE_METHODS(Task),
+
+	/* nodes with only output support */
+	DEFINE_NODE_METHODS(MultiTreeRoot),
+	DEFINE_NODE_METHODS(MultiProject),
+	DEFINE_NODE_METHODS(MultiCollect),
+	DEFINE_NODE_METHODS(MultiSelect),
+	DEFINE_NODE_METHODS(MultiTable),
+	DEFINE_NODE_METHODS(MultiJoin),
+	DEFINE_NODE_METHODS(MultiPartition),
+	DEFINE_NODE_METHODS(MultiCartesianProduct),
+	DEFINE_NODE_METHODS(MultiExtendedOp)
+};
+#endif
+
+void
```

Comment:
Hm, can we add a StaticAssert verifying that ExtensibleNodeMethods has the same length as CitusNodeTagNamesD?